OutOfSchool/OutOfSchool.ExternalFileStore.Impl/NotificationImplementations/MinioRedisNotificationBridge.cs

```diff
@@ -0,0 +1,130 @@
+﻿using Microsoft.Extensions.Logging;
+using Minio;
+using Minio.DataModel.Args;
+using Minio.DataModel.Notification;
+using OutOfSchool.ExternalFileStore.Models;
+using OutOfSchool.ExternalFileStore.NotificationInterfaces;
+using StackExchange.Redis;
+using System.Text.Json;
+
+namespace OutOfSchool.ExternalFileStore.NotificationImplementations;
+public class MinioRedisNotificationBridge : IDisposable
+{
+    private readonly MinioClient _minioClient;
+    private readonly IConnectionMultiplexer _redisConnection;
+    private readonly ILogger<MinioRedisNotificationBridge> _logger;
+    private readonly Dictionary<string, IDisposable> _subscriptions = new Dictionary<string, IDisposable>();
+    private readonly IProcessNotificationService _processNotificationService;
+
+    public MinioRedisNotificationBridge(
+            MinioClient minioClient,
+            IConnectionMultiplexer redisConnection,
+            ILogger<MinioRedisNotificationBridge> logger,
+            IProcessNotificationService processNotificationService)
+    {
+        _minioClient = minioClient ?? throw new ArgumentNullException(nameof(minioClient));
+        _redisConnection = redisConnection ?? throw new ArgumentNullException(nameof(redisConnection));
+        _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        _processNotificationService = processNotificationService ?? throw new ArgumentNullException(nameof(processNotificationService));
+    }
+
+    public async Task StartBridgeAsync(string bucket, NotificationFilter filter)
+    {
+        string subscriptionKey = $"{bucket}:{filter.Prefix}:{filter.Suffix}";
+
+        if (_subscriptions.ContainsKey(subscriptionKey))
+        {
+            _logger.LogWarning("Bridge already active for bucket {Bucket} with prefix {Prefix} and suffix {Suffix}",
+                bucket, filter.Prefix, filter.Suffix);
+            return;
+        }
+
+        _logger.LogInformation("Starting MinIO to Redis notification bridge for bucket {Bucket}", bucket);
+
+        try
+        {
+            var events = new List<EventType>
+                {
+                    EventType.ObjectCreatedAll
+                };
+
+            var args = new ListenBucketNotificationsArgs()
+                .WithBucket(bucket)
+                .WithEvents(events)
+                .WithPrefix(filter.Prefix)
+                .WithSuffix(filter.Suffix);
+
+            // Get Redis publisher
+            var publisher = _redisConnection.GetDatabase();
+            string channelName = $"minio-notifications:{bucket}:{filter.Prefix}:{filter.Suffix}";
+
+            // Subscribe to MinIO notifications
+            var observable = _minioClient.ListenBucketNotificationsAsync(args);
```

Comment:
This is great approach in general, but for our case when nothing is certain - we are linking two external dependencies (minio & redis) and we cannot be sure if they are going to be active at the same time. So `ListenBucketNotificationsAsync` should be completely independent of Redis flow even if there are some logic duplications.